### PR TITLE
Visual inspection update

### LINKED ISSUE #391: Visual Inspection
- [x] Test whether data matrix is actually only data
- [x] Automatic columnnames
- [ ] sparsly shift
- [x] Error bug -> better description, correctly update

### PR DESCRIPTION
**Pull request by LeaSeep**
closes #391

### CODE REVIEW

**Comment by PaulJonasJost** [comment]
> closes #391

Would be good to have a short description of what actually is supposed to have changed explicitly.

File: program/shinyApp/server.R [hunk 1]
```diff
@@ -510,25 +537,54 @@ server <- function(input,output,session){
...
+
       tryCatch(
         expr = {
           Matrix <- read_file(input$data_matrix1$datapath, check.names=T)
           Matrix2 <- read_file(input$data_matrix1$datapath, check.names=F)
           flag_csv <- T
         },
-        error = function(){
+        error = function(e){
           print("Not a real csv file!")
```
Comment (PaulJonasJost) [inline program/shinyApp/server.R:553]:
```suggestion
```

**Review by PaulJonasJost — CHANGES_REQUESTED** [review]
Will review more after it is working. Currently Testdata does not work anymore!

File: program/shinyApp/server.R [hunk 2]
```diff
@@ -577,22 +636,207 @@ server <- function(input,output,session){
...
+               grepl(snippetYes,check2) & 
+               grepl(snippetYes,check3) & 
+               grepl(snippetYes,check4) & # not crucial
+              # grepl(snippetYes,check5) & # not crucial
+               grepl(snippetYes,check6) &
+               grepl(snippetYes,check7)){
+              res_tmp[[session$token]]$passedVI <<- T
+            } else {
+              res_tmp[[session$token]]$passedVI <<- F
+            }
```
Comment (PaulJonasJost) [inline program/shinyApp/server.R:806]: I think this check also belongs to line 555. Otherwise the test always fail

**Review by PaulJonasJost — APPROVED** [review]
Little tweak then its fine i think.

Two things:
- I changed valid column names in the sample annotation and it sparked no error (do we want that/ not really care about it, which I think is valid)
- The data matrix I uploaded was not displayed. When switching to develop it was displayed upon uploading, something seems to have changed specifically for the data matrix display

**Review by PaulJonasJost — COMMENTED** [review]
Also was able to test the "fix the data" table. Small comment on this:
- It changes the EntrezID rownames, which should be no problem unless people have not had the entrezID somewhere else. Copying the original rownames to a column called "orig_rownames" would be cool
- I had a blank in my samplenames. This was "fixed" in the colnames of the matrix but not fixed in the rownames of the sample annotation leading to a new error

File: program/shinyApp/server.R (hunk 2 above)
Comment (LeaSeep) [reply program/shinyApp/server.R:835]: not sure what you ment, did insert it though

**Pull request merged**